core/src/main/scala/scalaz/std/String.scala

```diff
@@ -48,9 +48,8 @@ trait StringFunctions {
 
   // Parsing functions.
   private def intChars(s: String): Boolean = (s.length > 0) && (s.forall(c => c.isDigit || c == '-'))
```

Comment:
If you will, `intChars` would need to be fixed, to accept `-` only if it appears as first char of a multi chars string.
I think a correct impl could be 
`s.length == 1 && s.charAt(0).isDigit || ((s.length > 1) && s.stripPrefix("-").forall(_.isDigit))`

Reply:
Done